### PR TITLE
Fixes #15556: List all plugin in plugin status page

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15556

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala [hunk 1]
```diff
@@ -518,7 +522,35 @@ class Boot extends Loggable {
...
+
+    val nonScala = RudderConfig.jsonPluginDefinition.getInfo().either.runNow match {
+      case Left(err) =>
+        PluginLogger.error(s"Error when trying to read plugins index file '${RudderConfig.jsonPluginDefinition.index.pathAsString}': ${err.fullMsg}")
+        Nil
+      case Right(plugins) =>
+        val scalaKeys = scalaPlugins.keySet
+        // log parsing errors
+        plugins.collect { case Left(e) => e }.foreach { e =>
+          PluginLogger.error(s"Error when parsing loggin information from index file '${RudderConfig.jsonPluginDefinition.index.pathAsString}': ${e.fullMsg}")
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala]: "plugin information" no?

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+// this is only used internally but liftweb can't hand it if not top level
+protected final case class JsonPluginFile(plugins: Map[String, JValue])
+
+// this is only used internally but liftweb can't hand it if not top level
+protected final case class JsonPluginRaw(
+    name          : String
+  , version       : String
+  , files         : List[String]
+  , `jar-files`   : List[String]
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala]: why the ` ` ?

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  }
+
+  def parseJson(json: String) = {
+    parseFile(json) >>= decode
+  }
+
+  def getInfo(): IOResult[List[Either[RudderError, JsonPluginDef]]] = {
+    IOResult.effect(index.exists).flatMap { exists =>
+      if(exists) check *> readIndex() >>= parseJson
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala]: wow, i don't know this syntax

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala (hunk 2 above)
Comment (fanf) [reply webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala]: The? Backquote? It's to allow `-` in the name.

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala (hunk 3 above)
Comment (fanf) [reply webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala]: `*>` means "exec what's on left and discard result on success", and `>>=` means `flatMap` (and it binds the result to the next method param)

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala (hunk 2 above)
Comment (ncharles) [reply webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/RudderPluginJson.scala]: oh, ok

**Comment by fanf** [comment]
OK, squash merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**